File: Sway3i/src/main/java/com/sway3i/dto/Course/Response/CourseWithDetailsResponseDTO.java
package com.sway3i.dto.Course.Response;

import com.sway3i.dto.Utils.ProgramDTO;
import com.sway3i.dto.Utils.UserDTO;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CourseWithDetailsResponseDTO {

    private Long id;
    private LocalDate createdAt;
    private UserDTO createdByUser;
    private String courseName;
    private String subject;
    private String courseDetails;
    private String courseIsFor;
    private long price;
    private String city;
    private String educationLevel;
    private String type;
    private int maxStudents;
    private List<ProgramDTO> programs;

    private String link;
    private String address;

}

File: Sway3i-FrontEnd/src/app/dto/Course/responses/course-with-details-response-DTO.ts
import { ProgramDTO } from "../../Utils/program-DTO";
import { UserDTO } from "../../Utils/user-DTO";

export interface CourseWithDetailsResponseDTO {
  id: number;
  createdAt: string; // Date
  createdByUser: UserDTO;
  subject: string;
  courseName: string;
  courseDetails: string;
  courseIsFor: string;
  price: number;
  city: string;
  educationLevel: string;
  type: string;
  maxStudents: number;
  programs: ProgramDTO[];
  link: string;
  address: string;
}
